### PR TITLE
Core/output: commented dt limiters and add residual logs in hdf5

### PR DESCRIPTION
**Pull request by tduretz**

### CODE REVIEW

File: SOURCE/HDF5Output.c [hunk 1]
```diff
@@ -770,13 +770,19 @@ void WriteOutputHDF5( grid *mesh, markers *particles, surface *topo, markers* to
...
+    // Add cell flags for debugging 
     AddFieldToGroup_generic( _TRUE_, name, "Centers" , "BCp" , 'c', (model.Nx-1)*(model.Nz-1), mesh->BCp.type, 1 );
-
     AddFieldToGroup_generic( _TRUE_, name, "Flags", "tag_s" , 'c', (model.Nx-0)*(model.Nz-0), mesh->BCg.type,  1 );
     AddFieldToGroup_generic( _TRUE_, name, "Flags", "tag_n" , 'c', (model.Nx-1)*(model.Nz-1), mesh->BCp.type,  1 );
     AddFieldToGroup_generic( _TRUE_, name, "Flags", "tag_u" , 'c', (model.Nx-0)*(model.Nz+1), mesh->BCu.type,  1 );
     AddFieldToGroup_generic( _TRUE_, name, "Flags", "tag_v" , 'c', (model.Nx+1)*(model.Nz-0), mesh->BCv.type,  1 );
 
+    // Add non-iteration log
+    AddFieldToGroup_generic( _TRUE_, name, "Iterations", "rx_abs"          , 'd', Nmodel.nit_max+1, Nmodel.rx_abs         ,  1 );
```
Comment (kulakovri) [inline SOURCE/HDF5Output.c]: "name" - file_name, step_name?

File: SOURCE/HDF5Output.c [hunk 2]
```diff
@@ -770,13 +770,19 @@ void WriteOutputHDF5( grid *mesh, markers *particles, surface *topo, markers* to
...
+    AddFieldToGroup_generic( _TRUE_, name, "Iterations", "rz_abs"          , 'd', Nmodel.nit_max+1, Nmodel.rz_abs         ,  1 );
+    AddFieldToGroup_generic( _TRUE_, name, "Iterations", "rp_abs"          , 'd', Nmodel.nit_max+1, Nmodel.rp_abs         ,  1 );
+    AddFieldToGroup_generic( _TRUE_, name, "Iterations", "LogIsNewtonStep" , 'i', Nmodel.nit_max+1, Nmodel.LogIsNewtonStep,  1 );
```
Comment (kulakovri) [inline SOURCE/HDF5Output.c]: As far as I can see "Iterations" and other group names can be initialised only once and reused
Can we create HDF5Output.h and define those constants there?

File: SOURCE/HDF5Output.c [hunk 3]
```diff
@@ -770,13 +770,19 @@ void WriteOutputHDF5( grid *mesh, markers *particles, surface *topo, markers* to
...
+    AddFieldToGroup_generic( _TRUE_, name, "Iterations", "NumberSteps"     , 'i',                1, Nmodel.nit            ,  1 );
```
Comment (kulakovri) [inline SOURCE/HDF5Output.c]: dtype also can be stored in a constant

File: SOURCE/HDF5Output.c [hunk 4]
```diff
@@ -770,13 +770,19 @@ void WriteOutputHDF5( grid *mesh, markers *particles, surface *topo, markers* to
...
-
     AddFieldToGroup_generic( _TRUE_, name, "Flags", "tag_s" , 'c', (model.Nx-0)*(model.Nz-0), mesh->BCg.type,  1 );
     AddFieldToGroup_generic( _TRUE_, name, "Flags", "tag_n" , 'c', (model.Nx-1)*(model.Nz-1), mesh->BCp.type,  1 );
     AddFieldToGroup_generic( _TRUE_, name, "Flags", "tag_u" , 'c', (model.Nx-0)*(model.Nz+1), mesh->BCu.type,  1 );
     AddFieldToGroup_generic( _TRUE_, name, "Flags", "tag_v" , 'c', (model.Nx+1)*(model.Nz-0), mesh->BCv.type,  1 );
 
+    // Add non-iteration log
+    AddFieldToGroup_generic( _TRUE_, name, "Iterations", "rx_abs"          , 'd', Nmodel.nit_max+1, Nmodel.rx_abs         ,  1 );
+    AddFieldToGroup_generic( _TRUE_, name, "Iterations", "rz_abs"          , 'd', Nmodel.nit_max+1, Nmodel.rz_abs         ,  1 );
+    AddFieldToGroup_generic( _TRUE_, name, "Iterations", "rp_abs"          , 'd', Nmodel.nit_max+1, Nmodel.rp_abs         ,  1 );
```
Comment (kulakovri) [inline SOURCE/HDF5Output.c]: What does "_generic" say about function? 
I cannot find any other AddFieldToGroup functions here

File: SOURCE/HDF5Output.c [hunk 5]
```diff
@@ -770,13 +770,19 @@ void WriteOutputHDF5( grid *mesh, markers *particles, surface *topo, markers* to
...
     AddFieldToGroup_generic( _TRUE_, name, "Centers" , "BCp" , 'c', (model.Nx-1)*(model.Nz-1), mesh->BCp.type, 1 );
-
     AddFieldToGroup_generic( _TRUE_, name, "Flags", "tag_s" , 'c', (model.Nx-0)*(model.Nz-0), mesh->BCg.type,  1 );
     AddFieldToGroup_generic( _TRUE_, name, "Flags", "tag_n" , 'c', (model.Nx-1)*(model.Nz-1), mesh->BCp.type,  1 );
     AddFieldToGroup_generic( _TRUE_, name, "Flags", "tag_u" , 'c', (model.Nx-0)*(model.Nz+1), mesh->BCu.type,  1 );
     AddFieldToGroup_generic( _TRUE_, name, "Flags", "tag_v" , 'c', (model.Nx+1)*(model.Nz-0), mesh->BCv.type,  1 );
 
+    // Add non-iteration log
+    AddFieldToGroup_generic( _TRUE_, name, "Iterations", "rx_abs"          , 'd', Nmodel.nit_max+1, Nmodel.rx_abs         ,  1 );
+    AddFieldToGroup_generic( _TRUE_, name, "Iterations", "rz_abs"          , 'd', Nmodel.nit_max+1, Nmodel.rz_abs         ,  1 );
```
Comment (kulakovri) [inline SOURCE/HDF5Output.c]: What is "dim" argument? I did notice that is always used as 1
Do we need it?

Comment (kulakovri) [reply SOURCE/HDF5Output.c]: Same goes for the 'compress' argument. Do we need it?
First argument should be the most important, but that one looks like an argument, that is used only for development

**Review by kulakovri — CHANGES_REQUESTED** [review]

**Comment by kulakovri** [comment]
Please resolve conflict with the master branch

File: SOURCE/HDF5Output.c (hunk 1 above)
Comment (tduretz) [reply SOURCE/HDF5Output.c]: changed to FileName

File: SOURCE/HDF5Output.c (hunk 3 above)
Comment (tduretz) [reply SOURCE/HDF5Output.c]: OK, changed `char d_type` to `const char d_type`

File: SOURCE/HDF5Output.c (hunk 4 above)
Comment (tduretz) [reply SOURCE/HDF5Output.c]: Nothing, these functions are inherited from previous codes (not written by me initially). Style could be adjusted to better suit the frame of MD7.0.

File: SOURCE/HDF5Output.c (hunk 5 above)
Comment (tduretz) [reply SOURCE/HDF5Output.c]: We could need dim at some point, if we decide to store velocity as vectors (dim=2) for example.

We do not need `compress`, I've just removed it.

**Review by kulakovri — CHANGES_REQUESTED** [review]
Need to resolve conflict

**Pull request merged**